axelrod/tests/unit/test_player.py

```diff
@@ -20,7 +20,9 @@ class TestPlayerClass(unittest.TestCase):
 
     name = "Player"
```

Comment:
I'm just suggesting you test the functionality for in instance of the
`Player` class. So that upon initialisation of the `Player` class it's
behaviour dictionary is the same as the default dictionary.

I think it's a pretty trivial test but at least it checks that the
initialisation dictionary doesn't do something weird for the `Player`
class. I think I'm suggesting something like:

```
self.assertEqual(player.behaviour, default_behaviour) # we'd just need to
grab/create default_behaviour
```

For each inherited class, as you said above I agree we want to test them
individually.

On Tue, Sep 8, 2015 at 2:17 PM Owen Campbell <notifications@github.com>
wrote:

> In axelrod/tests/unit/test_player.py
> <https://github.com/Axelrod-Python/Axelrod/pull/293#discussion_r38923423>:
>
> > @@ -20,7 +20,9 @@ class TestPlayerClass(unittest.TestCase):
> >
> >      name = "Player"
>
> Tricky to do - anything you write needs to take into account the fact that
> the dictionary will be overridden by all the inherited classes and, unless
> they have all dimensions specified, you will hit missing key errors.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Axelrod-Python/Axelrod/pull/293/files#r38923423>.
>